src/content/docs/react/event-handling.mdx

```diff
@@ -1,3 +1,446 @@
+
+## Common User Events in React
+
+React supports several standard events that you can handle. Here are some commonly used ones:
+
+- **onClick**: Triggered when an element is clicked.
+- **onChange**: Used to handle changes in input fields.
+- **onSubmit**: Executes when a form is submitted.
+- **onKeyDown**, **onKeyUp**, **onKeyPress**: Handle keyboard events.
+- **onFocus** and **onBlur**: The `onFocus` event is triggered when an input gains focus, and `onBlur` is triggered when it loses focus.
+- **onMouseEnter** and **onMouseLeave**: These events handle mouse hover interactions.
+- **onDoubleClick**: The onDoubleClick event is triggered when an element is double-clicked.
+- **onScroll**: The `onScroll` event is triggered when an element is scrolled.
+
+These events provide an efficient way to manage user interactions in the app.
+
+### Example of every event
+
+<Tabs>
+  <TabItem label="onClick">
+    <Code code={onclick} lang="jsx" title="OnClick.jsx" />
+  </TabItem>
+   <TabItem label="onChange">
+    <Code code={onchange} lang="jsx" title="OnChange.jsx" />
+  </TabItem>
+   <TabItem label="onSubmit">
+    <Code code={onsubmit} lang="jsx" title="OnSubmit.jsx" />
+  </TabItem>
+   <TabItem label="KeyEvents">
+    <Code code={onkey} lang="jsx" title="OnKey.jsx" />
+  </TabItem>
+  <TabItem label="onFocus&onBlur">
+    <Code code={onfocus} lang="jsx" title="onFocus&onBlur.jsx" />
+  </TabItem>
+  <TabItem label="MouseEvents">
+    <Code code={onmouse} lang="jsx" title="MouseEvents.jsx" />
+  </TabItem>
+  <TabItem label="onScroll">
+    <Code code={onscroll} lang="jsx" title="onscroll.jsx" />
+  </TabItem>
+</Tabs>
+
+## Event Binding and Event Handlers
+
+In React, handling user interactions is crucial for creating dynamic applications. This section discusses the concept of event binding, how to define and attach event handlers to elements, and best practices for effective event handling.
+
+### What is Event Binding?
+
+Event binding refers to the process of connecting an event (such as a click or a change) to a specific handler function that executes when that event occurs. In React, event handlers are usually defined as methods within a component and can be attached directly to elements via JSX attributes.
+
+### Defining Event Handlers
+
+Event handlers can be defined as methods within a class component or as functions within a functional component. Here's how to create event handlers in both types of components:
+#### 1. Class Components
+
+In class components, you need to bind the event handler to the class instance, typically done in the constructor or by using class fields.
+
+<Tabs>
+  <TabItem label="Dependency">
+    Install the following libraries. Assuming you have installed
+    <Code code={`import React from 'react'`} lang="shell" />
```

Comment:
This is not a dependency. dependency is nothing but libraries that you need to install.